lib/replica_pools/connection_proxy.rb

```diff
@@ -64,8 +64,8 @@ def with_leader
       end
     end
 
-    def transaction(*args, &block)
-      with_leader { leader.transaction(*args, &block) }
+    def transaction(...)
+      with_leader { leader.transaction(...) }
```

Comment:
Will this work on Ruby 2.7, or is this intended for post-upgrade once we're fully on Ruby 3.0?

Reply:
This will work on ruby 2.7, the other pattern, `(arg, ...)` is not supported until ruby 3.0